lib/r10k/git/rugged/credentials.rb

```diff
@@ -125,4 +143,49 @@ def get_git_username(url, username_from_url)
 
     user
   end
+
+  def github_app_token(app_id, private_key, ttl)
+    raise R10K::Git::GitError, _('App id contains invalid characters.') unless app_id =~ /^\d+$/
+    raise R10K::Git::GitError, _('Token ttl contains invalid characters.') unless ttl =~ /^\d+$/
+    raise R10K::Git::GitError, _('App key is missing or unreadable') unless File.readable?(private_key)
+
+    begin
+      ssl_key = OpenSSL::PKey::RSA.new(File.read(private_key).strip)
+      ssl_key.private?
+    rescue OpenSSL::PKey::RSAError
+      raise R10K::Git::GitError, _('App key is not a valid SSL private key')
+    end
+
+    logger.debug2 _("Using Github App token from %{token_path} and id %{app_id}") % { token_path: private_key, app_id: app_id }
+
+    payload = { iat: Time.now.to_i, exp: Time.now.to_i + ttl.to_i, iss: app_id }
```

Comment:
Might be good to put the `Time.now.to_i` bit into a variable, so we have a consistent notion of "now" throughout this method.